README.md

```diff
@@ -314,52 +315,95 @@ module.exports = {
 
-Type: `string`. Default: `''`.
+- **Exporting a Function (Preferred)**: You can export a function which would then be used as a factory method for your [configuration object](#configuration-object-properties). This function is called the following parameter properties of an object:
 
-Root-relative path to the base directory on the domain where the site will be deployed.
+```javascript
+// underreact.config.js
+/**
+ * @param {Object} opts 
+ * @param {Webpack} opts.webpack - The Webpack object.
+ * @param {'start'|'build'|'serve-static'} opts.command - The current command Underreact is following.
+ * @param {'production'|'development'} opts.mode - The current mode of Underreact.
+ */
+module.exports = function({ webpack, command, mode }) {
+  return {/*Underreact configuration object*/}
+}
+```
 
-There's a good chance your app isn't at the root of your domain. So this option represents the path of your site *within* that domain.
+- **Exporting an Object**: You can also directly export the [configuration object](#configuration-object-properties). This is a great way to start experimenting with Underreact's configuration. For example in the code below we simply want to modify the `siteBasePath`:
 
-For example, if your app is at `https://www.special.com/ketchup/*`, you should set `siteBasePath: ketchup`.
+```javascript
+// underreact.config.js
+module.exports = {
+   siteBasePath: 'fancy'
+}
+```
 
-### stylesheets
+## Configuration Object Properties
 
-Type: `Array<string>`. Absolute paths, please. Default: `[]`.
+### devServerHistoryFallback
 
-An array of filenames pointing to stylesheets that you want to include in your site.
+Type: `boolean`. Default: `false`.
 
-These will be processed by PostCSS with [Autoprefixer](https://github.com/postcss/autoprefixer) and any other [postcssPlugins](#postcssplugins) that you specify; concatenated in the order specified; and added to the `<head>` of your document.
+Set to `true` if you want to use HTML5 History for client-side routing (as opposed to hash routing). This configures the development server to fall back to `index.html` when you request nested routes.
 
-Assets referenced by your stylesheets will be hashed and copied to the [outputDirectory](#outputdirectory), unless they are absolute URLs.
+This is `false` by default because it should only be *intentionally* turned on, when you know you're going to configure your server to allow for HTML5 History—powered client-side routing.
 
-### vendorModules
+### htmlSource
 
-Type: `Array<string>`. Default: `[]`.
+Type: `string`. Absolute path, please. Default: `${project-directory}/src/html.js`.
 
-Identifiers of npm modules that you want to be added to the vendor bundle.
-The purpose of the vendor bundle is to deliberately group dependencies that change relatively infrequently — so this bundle will stay cached for longer than the others.
+The path to your HTML source file. For more information, read [Defining your HTML](#defining-your-html).
```

Comment:
A convention I've been trying to follow is to wrap section titles in quotation marks.